src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -51,46 +52,121 @@ internal sealed class TupleTypeSymbol : NamedTypeSymbol, ITupleTypeSymbol
             out ImmutableArray<TupleFieldSymbol> fields
             )
         {
-            NamedTypeSymbol underlyingType;
-            fields = ImmutableArray<TupleFieldSymbol>.Empty;
+            Debug.Assert(elementNames.IsDefaultOrEmpty || elementTypes.Length == elementNames.Length);
+            Debug.Assert(elementTypes.Length > 1 && elementTypes.Length < 8); // PROTOTYPE
 
-            switch (elementTypes.Length)
+            var tupleType = binder.GetWellKnownType(EnumType(elementTypes.Length), diagnostics, syntax);
+            NamedTypeSymbol underlyingType = tupleType.Construct(elementTypes);
```

Comment:
Ok, it looks like for all generic well-known/special symbols we assume constraints have the "right" shape. They are (the constraints) are not needed for analysis. If we feel like this is something we need to tighten, we should do this across the board.